dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataOkHttpUploader.kt

```diff
@@ -31,8 +31,12 @@ internal class DataOkHttpUploader(
     val androidInfoProvider: AndroidInfoProvider
 ) : DataUploader {
 
-    private var attempts = 1
+    private var attempts: Int = 1
```

Comment:
why not making this one as `@Volatile` or even `AtomicInteger`. There is no concurrency right now (so probably `AtomicInteger` is not quite useful), but for the changes at least shouldn't we declare volatile access?

Reply:
I was almost convinced that primitives are volatile in JVM. The docs are saying either true or false so let's do this also Volatile. There is no need for Atomicity there.

Reply:
https://docs.oracle.com/javase/tutorial/essential/concurrency/atomic.html

> Reads and writes are atomic for reference variables and for most primitive variables (all types except long and double).
Reads and writes are atomic for all variables declared volatile (including long and double variables).

I'm not sure about ART though, but maybe it should maintain this as well.

Reply:
Also atomic read/write is different from cross-thread visibility though.

> When a field is declared volatile, the compiler and runtime are put on notice that this variable is shared and that operations on it should not be reordered with other memory operations. Volatile variables are not cached in registers or in caches where they are hidden from other processors, so a read of a volatile variable always returns the most recent write by any thread.

Reply:
Yes there are 2 different things but I am pretty sure I've read also about volatility once, anyway I guess without going into the bytecode you will never now for sure.